pkg/controller/deployment/progress.go

```diff
@@ -47,7 +47,7 @@ func (dc *DeploymentController) syncRolloutStatus(allRSs []*apps.ReplicaSet, new
 	isCompleteDeployment := newStatus.Replicas == newStatus.UpdatedReplicas && currentCond != nil && currentCond.Reason == util.NewRSAvailableReason
 	// Check for progress only if there is a progress deadline set and the latest rollout
 	// hasn't completed yet.
-	if d.Spec.ProgressDeadlineSeconds != nil && !isCompleteDeployment {
+	if !util.NoProgressDeadline(d) && !isCompleteDeployment {
```

Comment:
Nit: could we invert the function to `HasProgressDeadline` to avoid the double negative?

Reply:
Good point. Done.